src/test/java/com/vackosar/gitflowincrementalbuild/integration/MavenIntegrationTestBase.java

```diff
@@ -129,6 +130,10 @@ void initialInstall(TestInfo testInfo) throws Exception {
                 INITIAL_INSTALL_DONE.add(cacheKey);
             }
         }
+        // this info is independent of the concrete test class, so no "mapped flag" required
+        if (currentMavenAffectedByMNG6173 == null) {
+            currentMavenAffectedByMNG6173 = executeBuild(false, false, "-version").contains("Apache Maven 3.3.9");
+        }
```

Comment:
If i understand this parameter correctly, it builds a switch for testing using maven 3.3.9 or testing on other maven versions.
Idea: execute maven version specific tests, instead of parameterize them to test different things depending on used maven. Maybe with a version-precondition (https://junit.org/junit5/docs/current/user-guide/#extensions-conditions) for a 3.3.9-specific test. But out of scope for this change.